### PR TITLE
Fix false positive for literal text in `if` conditions

### PR DESCRIPTION
**Pull request by ericsciple**
Follow-up to PR:
- https://github.com/actions/languageservices/pull/216

Related issue:
- https://github.com/github/vscode-github-actions/issues/542

### Summary

Use `token.value` (parsed string without YAML quotes) instead of `token.source` (raw YAML text) for expression parsing in single-line strings. This fixes a false positive where `if: "${{ expr }}"` incorrectly triggered the "literal text in condition" error because the outer quotes were treated as literal text.

### CODE REVIEW

File: PR_REVIEW.md [hunk 1]
```diff
@@ -1,59 +0,0 @@
-# PR #283 Review: Use property descriptions for completion items
```
Comment (ericsciple) [inline PR_REVIEW.md]: This file was checked in on accident

File: workflow-parser/src/templates/template-reader.ts [hunk 2]
```diff
@@ -496,14 +502,17 @@ class TemplateReader {
         );
 
         let tr = token.range!;
-        if (tr.start.line === tr.end.line) {
-          // If it's a single line expression, adjust the range to only cover the sub-expression
+        if (isSingleLine) {
+          // Single-line: adjust the range to only cover the sub-expression.
+          // Calculate offset to account for YAML quote characters.
+          // For example, `"${{ expr }}"` has source with quotes, value without.
+          const offset = (token.source ?? raw).indexOf(OPEN_EXPRESSION) - raw.indexOf(OPEN_EXPRESSION);
```
[the remark this answers is not in the document]
Comment (ericsciple) [reply workflow-parser/src/templates/template-reader.ts]: The concern is partially valid, but not for the reason stated.

For simple quoted strings like `"${{ a }} text ${{ b }}"`, the offset calculation is actually correct. The offset represents the fixed shift caused by the leading YAML quote character (1 for `"` or `'`), which applies equally to all positions in the string regardless of which expression is being processed.

However, there is a limitation when **YAML escape sequences** appear before subsequent expressions. For example:

```yaml
run: '${{ \'x\' }} ${{ y }}'
```

Here `\'x\'` in source (4 chars) becomes `'x'` in value (3 chars). The offset for the second `${{ y }}` would be slightly off because we don't account for the length difference introduced by the escape sequence earlier in the string.

Properly fixing this would require reimplementing YAML string escaping rules (which vary between single-quoted and double-quoted strings). Given that:
- The error is still correctly detected
- Only the squiggly underline position is affected
- The edge case is rare (YAML escapes before subsequent expressions)

I'm accepting this as a known limitation rather than adding significant complexity.

File: languageservice/src/validate.expressions-literal-text.test.ts [hunk 3]
```diff
@@ -211,4 +211,85 @@ jobs:
...
+`;
+      const result = await validate(createDocument("wf.yaml", input));
+
+      expect(result).toContainEqual(
+        expect.objectContaining({
+          code: "expression-literal-text-in-condition"
+        })
+      );
+    });
+  });
```
[the remark this answers is not in the document]
Comment (ericsciple) [reply languageservice/src/validate.expressions-literal-text.test.ts]: Added case with two expressions separated by literal text

**Review by rentziass — APPROVED** [review]
